src/reflection.ts

```diff
@@ -244,124 +244,136 @@ export class Parser {
-        } else if (elementType === reflection.BaseType.Obj) {
-          const vectorLambda = this.readVectorOfTablesLambda(field);
-          const subTableLambda = this.toObjectLambda(fieldType.index(), readDefaults);
-          lambdas[fieldName] = (t: Table) => {
-            const vector = vectorLambda(t);
-            if (vector === null) {
-              return null;
-            }
-            const result = [];
-            for (const table of vector) {
-              result.push(subTableLambda(table));
-            }
-            return result;
-          };
-        } else if (elementType === reflection.BaseType.Union) {
-          // Similar to how a union is serialized in a table, a vector of union gets a sidecar field
-          // that is an array of discriminator values. This sidecar field is named as the union field
-          // with _type suffix
-          //
-          // https://flatbuffers.dev/md__schemas.html
-
-          const discriminatorFieldName = `${fieldName}_type`;
-          const unionDiscriminator = this.getField(discriminatorFieldName, typeIndex);
-
-          lambdas[fieldName] = this.readVectorOfUnionsLambda(
-            field,
-            unionDiscriminator,
-            readDefaults,
-          );
-        } else {
-          throw new Error("Vectors of Arrays are not supported.");
-        }
-      } else if (baseType === reflection.BaseType.Union) {
-        // A union gets a sidecar field that is the discriminator value. This sidecar field is named
-        // as the union field with _type suffix.
+      } else if (elementType === reflection.BaseType.Union) {
+        // Similar to how a union is serialized in a table, a vector of union gets a sidecar field
+        // that is an array of discriminator values. This sidecar field is named as the union field
+        // with _type suffix
         //
         // https://flatbuffers.dev/md__schemas.html
 
         const discriminatorFieldName = `${fieldName}_type`;
         const unionDiscriminator = this.getField(discriminatorFieldName, typeIndex);
 
-        lambdas[fieldName] = this.readUnionLambda(field, unionDiscriminator, readDefaults);
-      } else if (baseType === reflection.BaseType.Array) {
-        if (!schema.isStruct()) {
-          throw new Error("Arrays are only supported inside structs, not tables");
-        }
-        const elementType = fieldType.element();
-        if (isScalar(elementType)) {
-          lambdas[fieldName] = this.readArrayOfScalarsLambda(field);
-        } else if (elementType === reflection.BaseType.Obj) {
-          lambdas[fieldName] = this.readArrayOfStructsLambda(field, readDefaults);
-        } else {
-          throw new Error("Arrays may contain only scalar or struct fields");
-        }
+        return this.readVectorOfUnionsLambda(field, unionDiscriminator, readDefaults);
       } else {
-        throw new Error("Unions are not supported in field " + field.name());
```

Comment:
I think this error was wrong now as we added union support. I changed it to return undefined but we could do something else if desired.

Reply:
Yeah I thought I saw this in the change and convinced myself it was right (like it was only going to happen for structs or something), but it's definitely wrong.

Are there any conditions (outside of an invalid schema) where we'd actually expect to hit the `return undefined` at this point? Tempted to suggest still throwing an error to bubble up issues rather than silently hiding that we failed to deserialize a field.

Reply:
I guess the question is whether for an unsupported feature it is better to still deserialize the _rest_ of the message as much as possible, or is it better to crash? I could see some consumers (e.g. foxglove users) who would rather see _most_ of their message than none at all.

Reply:
I don't mind merging it as-is if you have a strong preference for this, but I think my impression is that at this point we aren't specifically aware of features that aren't supported[^1], and so I'd mostly expect this to happen with a malformed schema or if there were a bug in this code.

My personal philosophy tends to be that when using something like foxglove I'd generally much rather have the tool yell at me that I am using an unsupported data type rather than have it present me with incorrect data and spending a bunch of time chasing ghosts as a consequence (and I would view failing to deserialize a given field as incorrect data). However, given that you all are the main users of this currently I don't really mind selecting for this behavior.

[^1]: Well, looks like we don't support `Vector64`.

Reply:
Of course, without the ability to patch foxglove myself I'd probably be a bit more irritated at foxglove chosing to not deserialize the message at all.

Reply:
Updated to `throw`!